compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -631,6 +626,24 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
             .any(|span_label| span_label.label.is_some() && span_label.span == span)
             && local_visitor.found_arg_pattern.is_none()
         {
+            let (kind_str, const_value) = match arg.unpack() {
+                GenericArgKind::Type(_) => ("type", None),
+                GenericArgKind::Const(_) => ("the value", Some(())),
+                GenericArgKind::Lifetime(_) => bug!("unexpected lifetime"),
+            };
+
+            if let Some(suggestion) = const_value
```

Comment:
This should work, but if you look further down in the method, there are lots of special cases for missing type parameters designed to emit good error messages in a variety of situations. Did you try modifying this code, so we can emit const errors in exactly the same way as type errors?

Reply:
I tried to do this, but the `annotate_method` method (where the errors for generics are emitted) is not even called for uninferred const generics.

Reply:
@Stupremee: do you know what the issue is? I would have thought you could tweak the methods to work with consts the same way they work with types.

Reply:
Sorry I forgot this PR in the last few days. The last time I tried the problem was, that the `annotate_method` method is never called for uninferred const arguments

Reply:
I think they ought to be. Could you work out why they're behaving differently. Const arguments should trigger `annotate_method` too.

Reply:
It doesn't get called because both of these conditions will evaluate to `false` / `None`

https://github.com/rust-lang/rust/blob/d1df5ef96724e44c35628b7b8b6f25644b260fe3/compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs#L127
https://github.com/rust-lang/rust/blob/d1df5ef96724e44c35628b7b8b6f25644b260fe3/compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs#L140

Reply:
Okay, let me take a better look and get back to you.

Reply:
It will take more investigation than I have time for. If you address my other comments, and add a `FIXME(const_generics)` comment explaining that we'd like to integrate this with the handling for type arguments, that should suffice for now.